app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteAddFragment.kt

```diff
@@ -60,10 +60,17 @@ class AutocompleteAddFragment : Fragment() {
                     .toLowerCase()
                     .removePrefixesIgnoreCase("http://", "https://", "www.")
 
-            if (domain.isEmpty()) {
-                domainView.error = getString(R.string.preference_autocomplete_add_error)
-            } else {
-                saveDomainAndClose(activity.applicationContext, domain)
+            launch(UI) {
```

Comment:
Discussed on Slack:

>  I actually don’t think CustomDomains.load needs to be a suspending function: it’s not calling anything that suspends, afaict so I don’t think coroutines are necessary here
>
> That being said, using a suspending function on the UI thread creates a lot of complexity: you allow the UI thread to return to the system before your code completes. For example, one code snippet I had once… 

```kotlin
    // In a coroutine context...
     
    val screenshot = async { getScreenshot() }
    val title = async { getTitle() }
     
    screenshotView.drawable = screenshot.await() // <-- suspends here
    titleView.text = title.await() // <-- suspends here
     
    // Control returns to system at each suspension so ^ screenshot appears,
    // then title appears, and now we hide them: there's an unintended pop-in effect.
    screenshotView.alpha = 0
    titleView.alpha = 0
     
    screenshotView.animate()
    titleView.animate() 
```

> mcomella [3 minutes ago]
wrt my example, in this case, it looks like it’ll be **okay** but that may not be true if the code changes and in your code the user will have an opportunity to interact with the app (unexpectedly) before the fragment manager code gets to run (in the final case).

> I think a generally safer pattern is: 
```kotlin
    launch(CommonPool) {
        // Do suspending work here
        launch(UI) {
            // Do non-suspending UI work here to avoid accidentally giving control to system
        }
    } 
```

> Suspending on the UI thread also forces scheduling on the UI, hurting its performance but that could be negligible

Reply:
@mcomella Tracking auditing usage of suspend in `android-components` here: https://github.com/mozilla-mobile/android-components/issues/79. If the API changes then we can remove the async code here 👍 

I'll go ahead and address the other comments as well